### ISSUE TITLE
Update Unifi Controller

### ISSUE THREAD

**Issue by dedsecproyt** [mimugmail/opn-repo#156]
Pls make auto-updater for latest version of unifi controller

**Comment by mimugmail** [comment]
No, not now. The newest version updates mongo to 4.4 which will require manual backup and restore

**Comment by DrDrache** [comment]
Is there a possibility for an updated package with the caveats of manual backup/restore; for those of us who want to move to it, as well as new users? ex: a new package name to not force the upgrade for unknowing users.

new users (new to this package/plugin) would be better served with the more recent version, and users who choose can work with a backup/restore and migrate.

**Comment by mimugmail** [comment]
Yep, but it will take some weeks ..

**Comment by bkw777** [comment]
Is there a way to upgrade unify myself manually? Would following the directions from https://www.freshports.org/net-mgmt/unifi7/ work? assuming adjusting paths

**Comment by mimugmail** [comment]
I think you can install the FreeBSD pkg, it would also require mongo42

**Comment by bkw777** [comment]
I failed to figure it out.
Don't worry I'm not begging for any help recoving from a mess either, I'm already all back up now although what I tried did break stuff and require fixing.

Previously I have installed the generic unifi7 pkg in ordinary freebsd jails on truenas core and it works fine.

Here on opnsense I tried to enable the generic freebsd pkg repo by changing enable no to yes in /usr/local/etc/pkg/repos/FreeBSD.conf, then pkg update & pkg upgrade, which updated pkg itself to 1.20.4 and I was able to `pkg install mc` which is a package that isn't in the opnsense repo, so that seeming proves the generic repo was working. But then the opnsense repos didn't work any more and neither "pkg update" in ssh nor "check for updates" in the gui worked any more (for normal opnsense stuff). The gui also showed all installed plugins as orphaned, regardless which repo, and only showed the installed ones.
```
root@router:~ # pkg update
Updating FreeBSD repository catalogue...
FreeBSD repository is up to date.
Updating OPNsense repository catalogue...
pkg: No SRV record found for the repo 'OPNsense'
Fetching meta.conf: 100%    163 B   0.2kB/s    00:01    
pkg: packagesite URL error for pkg+http://mirrors.nycbug.org/pub/opnsense/FreeBSD:13:amd64/23.1/latest/packagesite.pkg -- pkg+:// implies SRV mirror type
pkg: packagesite URL error for pkg+http://mirrors.nycbug.org/pub/opnsense/FreeBSD:13:amd64/23.1/latest/packagesite.txz -- pkg+:// implies SRV mirror type
Unable to update repository OPNsense
Updating mimugmail repository catalogue...
Fetching meta.conf:   0%
Fetching packagesite.pkg:   0%
mimugmail repository is up to date.
Error updating repositories!
root@router:~ # 
```

I did `opnsense-bootstrap` to recover from that and that went fine.

That reset everything to 23.1, I re-updated to 23.7.b_183, reinstalled the mimugmail repo and reinstalled the unifi7 plugin from the gui, and amazingly I didn't even have to restore the unifi config from backup. I had backups and could also have just reinstalled everything from scratch but it wasn't needed in the end.

Before recovering I still tried to install the generic unifi7 package but couldn't get pkg to find it. It only showed the mimugmail one. I even tried uninstalling unifi7 and then disabling the mimugmail repo to get around anything that might have been hiding or prioritizing, but then it just didn't show any unifi7 package at all.

So if there is a way to do it, it requires something more than the obvious/apparent steps.

**Comment by mihakralj** [comment]
OPNsense repo and FreeBSD repo are overlapping each other and (I believe that) FreeBSD repo takes higher priority than OPNsense repo. If you would disable FreeBSD repo and re-run `pkg update`, things should return back to normal.

Instead of using a repo, why don't you install generic Unifi7 directly from its url using `pkg add` command (which is pulling package straight from the source) instead of `pkg install` command (which is hunting for the package in exposed repos)

**Comment by bkw777** [comment]
I confess I didn't know pkg could do that, nor ever saw any page with links to any individual .pkg files, but I just googled one up so, thank you, I'll try it next time I'm willing to experiment and have time for a wipe & restore again.

**Comment by kode54** [comment]
Unifi controller is throwing 404 errors even after I completely wiped its settings and reinstalled it.

**Comment by mimugmail** [comment]
You need to wait atound 5 min after startup

**Comment by wi23avb** [comment]
After upgrade to opnsense 23.7.11 and unexpected power failure, my unifi package nas been corrupted.

I uninstalled and deleted all the leftovers via SSH
find / -name unifi and tried to install it again.

Unfortunately the service fails jest after couple of seconds running. Manuał start or opnsense reboot does not work. I'm not able to see a backup restore page under port 8443 since the service keeps flapping to disable state.

No logs under unifi has shown an error. 

Is there any working steps that i should follow in order to restore functionality? Thank you

**Comment by Qhilm** [comment]
> After upgrade to opnsense 23.7.11 and unexpected power failure, my unifi package nas been corrupted.
> 
> I uninstalled and deleted all the leftovers via SSH find / -name unifi and tried to install it again.
> 
> Unfortunately the service fails jest after couple of seconds running. Manuał start or opnsense reboot does not work. I'm not able to see a backup restore page under port 8443 since the service keeps flapping to disable state.
> 
> No logs under unifi has shown an error.
> 
> Is there any working steps that i should follow in order to restore functionality? Thank you

Bit late to the party, but one reason for this can be that another service is using 8080. For example, CrowdSec use this port by default.

**Comment by rterbush** [comment]
Not clear from this discussion but is there a recovery path for this issue? I've just discovered that the unifi server is not starting and see the following logged to server.log every few seconds.

```
[2024-11-22T14:17:56,393-07:00] <mongo-db> WARN  mongo  - Stop listening to Mongo logs after process has exited
[2024-11-22T14:17:56,393-07:00] <mongo-db> INFO  mongo  - Database process stopped, code=134
[2024-11-22T14:17:59,646-07:00] <mongo-db> WARN  mongo  - Stop listening to Mongo logs after process has exited
[2024-11-22T14:17:59,647-07:00] <mongo-db> INFO  mongo  - Database process stopped, code=134
```
